BaseLib/ConfigTreeUtil.h

```diff
@@ -0,0 +1,86 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef BASELIB_CONFIGTREEUTIL
+#define BASELIB_CONFIGTREEUTIL
+
+#include "ConfigTreeNew.h"
+
+namespace BaseLib
+{
+
+/*! Manages a ConfigTree and the <tt>boost::property_tree</tt> it depends on.
+ *
+ * The whole purpose of this class is making the management of said dependency easy.
+ */
+class ConfigTreeTopLevel final
+{
+public:
+    /*! Construct a new instance from the given data.
+     *
+     * \param filepath stored for use in error/warning messages
+     * \param be_ruthless if true, then warnings will raise errors, .i.e. lead to program abortion,
+     *                    else warnings will only warn
+     * \param ptree the underlying ptree of the created ConfigTree
+     */
+    explicit ConfigTreeTopLevel(std::string const& filepath,
+                                bool const be_ruthless,
+                                ConfigTreeNew::PTree&& ptree);
+
+    //! Access the contained ConfigTree.
+    ConfigTreeNew const& operator*() const;
```

Comment:
Initially it was by purpose, but when writing `operator->()` I forgot about it. ^^
Actually I'd rather remove the non-const `operator->()` and add a comment. The idea of all of this is to prevent that the user calls `checkAndInvalidate()` on a passed around instance of the `ConfigTreeNew`, instead the respective method of this wrapper should be used.